src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -114,6 +132,8 @@ internal bool TryExport(T data)
                 return true; // enqueue succeeded
             }
 
+            this.HealthReporter?.ReportBatchProcessorDroppedCount(1, this.DroppedCountTags);
```

Comment:
given we already keep track of droppedCount here, this feels like better suited for ObservableCounter. Was that considered before?

Reply:
Yes - We discussed this during one of the SIG meetings

Decided to use Counter as

1) This should not be called many times so it should not have an high impact on perf. 
2) Having ObservableCounter will add additional complexity with respect to https://github.com/open-telemetry/opentelemetry-dotnet/pull/4289#discussion_r1134390550.

Separately, we also discussed the potential to revisit the existing DroppedCount as it is only logged during shutdown https://github.com/open-telemetry/opentelemetry-dotnet/blob/bf9f07dc0ff17a6e9c113c7f46fba50a46e11c00/src/OpenTelemetry/BatchExportProcessor.cs#L222.

Reply:
@reyang @cijothomas Have opened https://github.com/open-telemetry/opentelemetry-dotnet/pull/4409 with alternate approach to use ObservableCounter.